### PR TITLE
adds pyjsg

### PR DESCRIPTION
**Pull request by marimeireles**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Review by xhochy — CHANGES_REQUESTED** [review]
This package should be noarch. A failing build on Windows is fine.

**Comment by marimeireles** [comment]
Hey @xhochy.
If I add a skip for windows then I get the error seen [here](https://github.com/conda-forge/staged-recipes/pull/26464#issuecomment-2133848632), no?
And if I don't skip windows... my package won't build?

**Comment by marimeireles** [comment]
But I see there's a name conflict.
I will change that.

**Comment by marimeireles** [comment]
I hope names with hyphens are within accepted?
Best,

**Comment by marimeireles** [comment]
Hey @xhochy sorry for the ping but I don't understand how this package should be noarch when it doesn't run on windows?
Best,

**Comment by xhochy** [comment]
If a dependency is missing, there will be a solver failure notifying the user of it. If the package itself doesn't work on Windows, then add a `__unix` dependency to `run`.

File: recipes/pyjsg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+requirements:
+  host:
+    - python
+    - pbr
+    - pip
+  run:
+    - python
+    - antlr4-python3-runtime >=4.9.3,<4.10.dev0
+    - jsonasobj >=1.2.1
+    - __unix  # [unix]
```
Comment (marimeireles) [inline recipes/pyjsg/meta.yaml:27]: Is this correct?

**Comment by marimeireles** [comment]
Sorry @xhochy I've read the docs and added a `- __unix  # [unix]` to my `run` but it's still failing.
I still don't understand this new tag. I don't understand why I should have a noarch when this package won't run on windows.

**Comment by xhochy** [comment]
>  I don't understand why I should have a noarch when this package won't run on windows.

Because it is still architecture independent on non-Windows systems. This will make maintenance of the package much simpler

**Pull request merged**